muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/evaluation/EvaluationRESTService.java

```diff
@@ -1369,6 +1375,128 @@ public Response archiveWorkspaceAssessmentRequest(@PathParam("WORKSPACEUSERENTIT
+        payload.getFeedback());
+    }
+    return Response.ok(createRestModel(journalFeedback)).build();
+  }
+  
+  @GET
+  @Path("/workspaces/{WORKSPACEENTITYID}/students/{STUDENTENTITYID}/journalfeedback")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
+  public Response getWorkspaceStudentJournalFeedback(@PathParam("WORKSPACEENTITYID") Long workspaceEntityId, @PathParam("STUDENTENTITYID") Long studentEntityId) {
+    // Entities
+    UserEntity studentEntity = userEntityController.findUserEntityById(studentEntityId);
+    
+    if (studentEntity == null) {
+      return Response.status(Status.BAD_REQUEST)
+        .entity(String.format("Malformed student entity %s", studentEntity))
+        .build();
+    }
+    
+    WorkspaceEntity workspaceEntity = workspaceController.findWorkspaceEntityById(workspaceEntityId);
+    if (workspaceEntity == null) {
+      return Response.status(Status.NOT_FOUND)
+        .entity(String.format("Could not find workspace entity %d", workspaceEntityId))
+        .build();
+    }
+    // Access
+    if (!sessionController.getLoggedUserEntity().getId().equals(studentEntity.getId())) {
+      if (!sessionController.hasWorkspacePermission(MuikkuPermissions.VIEW_USER_EVALUATION, workspaceEntity)) {
+        return Response.status(Status.FORBIDDEN).build();
+      }
+    }
+    
+    WorkspaceJournalFeedback journalFeedback = evaluationController.findWorkspaceJournalFeedbackByStudentAndWorkspace(studentEntity.getId(), workspaceEntity.getId());
+    
+    if (journalFeedback == null) {
+      return Response.noContent().build();
+    }
+    return Response.ok(createRestModel(journalFeedback)).build();
+  }
+  
+  @DELETE
+  @Path("/workspaces/{WORKSPACEENTITYID}/students/{STUDENTENTITYID}/journalfeedback/{ID}")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
+  public Response deleteWorkspaceJournalFeedback(@PathParam("WORKSPACEENTITYID") Long workspaceEntityId, @PathParam("STUDENTENTITYID") Long studentEntityId, @PathParam("ID") Long feedbackId) {
+    if (!sessionController.hasEnvironmentPermission(MuikkuPermissions.ACCESS_EVALUATION)) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    
+    WorkspaceJournalFeedback journalFeedback = evaluationController.findWorkspaceJournalFeedbackByStudentAndWorkspace(studentEntityId, workspaceEntityId);
+    
+    if (journalFeedback == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+    evaluationController.deleteWorkspaceJournalFeedback(journalFeedback);
+
+    return Response.noContent().build();
+  
+  }
+  
+  private RestWorkspaceJournalFeedback createRestModel(WorkspaceJournalFeedback journalFeedback) {
+    RestWorkspaceJournalFeedback restFeedback = new RestWorkspaceJournalFeedback();
+    UserEntity creatorEntity = userEntityController.findUserEntityById(journalFeedback.getCreator());
+    User creator = null;
+    
+    if (creatorEntity != null) {
```

Comment:
Users' names can be resolved faster via...

`userEntityController.getName(userEntity)`